codec-ohttp/src/main/java/io/netty/incubator/codec/ohttp/OHttpClientCodec.java

```diff
@@ -83,6 +82,8 @@ void destroy() {
 
     private ByteBuf cumulationBuffer = Unpooled.EMPTY_BUFFER;
     private boolean destroyed;
+    private boolean decodeCalled;
+    private boolean producedMessage;
 
```

Comment:
You need to set these variables inside `decode()`, like you do for the server codec.

Reply:
Doh! Fixing it